Makefile

```diff
@@ -0,0 +1,63 @@
+#  Copyright (C) 2016 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+ifdef http_proxy
+BUILD_PROXY=--build-arg http_proxy=$(http_proxy)
+RUN_PROXY=--env http_proxy=$(http_proxy)
+endif
+
+ifdef https_proxy
+BUILD_PROXY+=--build-arg https_proxy=$(https_proxy)
+RUN_PROXY+=--env https_proxy=$(https_proxy)
+endif
+
+IMAGE_BUILDER=cc-osbuilder
+KERNEL_REPO=https://github.com/clearcontainers/linux.git
+
+DOCKER_RUN=docker run \
+			--runtime runc \
+			--privileged \
+			-v /dev:/dev \
+			$(RUN_PROXY) \
+			-ti \
+			-v $(CURDIR)/workdir:/osbuilder \
+			$(IMAGE_BUILDER)
+
+rootfs: docker-build workdir
+	$(DOCKER_RUN) rootfs
+
+
+image: docker-build rootfs workdir
+	$(DOCKER_RUN) image
+
+kernel: workdir/linux docker-build 
+	$(DOCKER_RUN) kernel
+
+docker-build: 
+	cd scripts; \
+	docker build $(BUILD_PROXY) -t $(IMAGE_BUILDER) . 
+
+#we only clone kernel if does not exit
+workdir/linux: workdir
+	@echo Clone container kernel from $(KERNEL_REPO);\
+	cd workdir; \
+	git clone --depth=1 $(KERNEL_REPO); \
+	cd linux; \
+	make clear_containers_defconfig;
+
+workdir:
+	mkdir -p workdir
```

Comment:
agree, but in that case I dont know a good place to provide the image, kernel and rootfs where they will be modifying  and using.  I would not like to use a radom /tmp/directory. 

What about document it, and add variable in case the use would like to use a different directory?